xcode/Subconscious/Shared/UIKit/BlockEditor/BlockEditorModel.swift

```diff
@@ -7,35 +7,1336 @@
+        model.blocks.blocks[indexUp] = blockUp
+        model.blocks.blocks.remove(at: indexDown)
+        
+        let indexPathUp = IndexPath(
+            row: indexUp,
+            section: BlockEditor.Section.blocks.rawValue
+        )
+        
+        let indexPathDown = IndexPath(
+            row: indexDown,
+            section: BlockEditor.Section.blocks.rawValue
+        )
+
+        // Mark unsaved
+        model.setSaveState(.unsaved)
+
+        return Update(
+            state: model,
+            change: .mergeBlockUp(
+                reconfigure: indexPathUp,
+                delete: indexPathDown,
+                requestEditing: blockUp.id
+            )
+        )
+    }
+    
+    static func editing(
+        state: Self,
+        id: UUID
+    ) -> Update {
+        var model = state
+        model.blocks.blocks = state.blocks.blocks.map({ block in
+            block.setEditing(block.id == id) ?? block
+        })
+        return Update(state: model)
+    }
+    
+    static func renderEditing(
+        state: Self,
+        id: UUID
+    ) -> Update {
+        guard let index = state.blocks.blocks.firstIndex(whereID: id) else {
+            Self.logger.log("block#\(id) no block found with ID. Doing nothing.")
+            return Update(state: state)
+        }
+        
+        let indexPath = IndexPath(
+            row: index,
+            section: BlockEditor.Section.blocks.rawValue
+        )
+        
+        var model = state
+        model.blocks.blocks = state.blocks.blocks.map({ block in
+            block.setEditing(block.id == id) ?? block
+        })
+        
+        return Update(
+            state: model,
+            change: .reconfigureCollectionItem(indexPath)
+        )
+    }
+    
+    static func blur(
+        state: Self,
+        id: UUID
+    ) -> Update {
+        var model = state
+        model.blocks.blocks = state.blocks.blocks.map({ block in
+            if block.id == id {
+                return block.setEditing(false) ?? block
+            }
+            return block
+        })
+        return Update(state: model)
+    }
+    
+    static func renderBlur(
+        state: Self,
+        id: UUID
+    ) -> Update {
+        guard let index = state.blocks.blocks.firstIndex(whereID: id) else {
+            Self.logger.log("block#\(id) no block found with ID. Doing nothing.")
+            return Update(state: state)
+        }
+        
+        let indexPath = IndexPath(
+            row: index,
+            section: BlockEditor.Section.blocks.rawValue
+        )
+        
+        var model = state
+        model.blocks.blocks = state.blocks.blocks.map({ block in
+            if block.id == id {
+                return block.setEditing(false) ?? block
+            }
+            return block
+        })
+        
+        return Update(
+            state: model,
+            change: .reconfigureCollectionItem(indexPath)
+        )
+    }
+
+    // TODO: re-implement
```

Comment:
I'm assuming these TODOs are out of scope for this PR since they're related to bulk selection.

Reply:
Yes, out of scope. The selection mode wasn't working properly. Removed commented-out code, and added issues to TODOs.